### PR TITLE
Unify `preds, target` input arguments for `text` metrics [1of2] bert, bleu, chrf, sacre_bleu, wip, wil

### PR DESCRIPTION
**Pull request by stancld**
## What does this PR do?

This PR unifies `preds, target` input arguments for `text metrics` + also unifies some minor related stuff

Part of #716

Fixes: 
- [x] `bert`
- [x] `bleu`
- [x] `chrf`
- [x] `sacre_bleu`
- [x] `wip`
- [x] `wil`

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@stancld made the metrics as a checkbox also here so you can check progress and could you also update the original issue?

**Comment by Borda** [comment]
still WIP? :rabbit:

**Comment by stancld** [comment]
> still WIP? 🐰 fyi, [#727 (review)](https://github.com/PyTorchLightning/metrics/pull/727#pullrequestreview-846383656)

Yes a bit, I'll need to handle breaking changes add some deprecation warnings as you pointed out in #727

**Review by Borda — CHANGES_REQUESTED** [review]
still need warnings and back compatibility :rabbit:

**Comment by stancld** [comment]
> still need warnings and back compatibility 🐰

Currently adding these... Though, still thinking about if we shouldn't go with `None` both for `predictions` and `preds`. 🤔

File: torchmetrics/functional/text/bert.py [hunk 1]
```diff
@@ -450,8 +451,10 @@ def _rescale_metrics_with_baseline(
 
 
 def bert_score(
-    predictions: Union[List[str], Dict[str, Tensor]],
-    references: Union[List[str], Dict[str, Tensor]],
+    preds: Union[List[str], Dict[str, Tensor]],
+    target: Union[List[str], Dict[str, Tensor]],
+    predictions: Union[None, List[str], Dict[str, Tensor]] = None,
+    references: Union[None, List[str], Dict[str, Tensor]] = None,
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:457]: the old one shall be at the very end, imagine that you have a user who uses only positional arguments, this will break for him... :rabbit:

File: torchmetrics/functional/text/bert.py [hunk 2]
```diff
@@ -478,10 +481,10 @@ def bert_score(
     This implemenation follows the original implementation from `BERT_score`_
 
     Args:
-        predictions:
+        preds:
             Either an iterable of predicted sentences or a `Dict[str, torch.Tensor]` containing `input_ids` and
             `attention_mask` `torch.Tensor`.
-        references:
+        target:
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:487]: let's also include the old names with a just warning that they are deprecated in favor of...

File: torchmetrics/functional/text/bert.py [hunk 3]
```diff
@@ -548,14 +551,29 @@ def bert_score(
 
     Example:
         >>> from torchmetrics.functional.text.bert import bert_score
-        >>> predictions = ["hello there", "general kenobi"]
-        >>> references = ["hello there", "master kenobi"]
-        >>> bert_score(predictions=predictions, references=references, lang="en")  # doctest: +SKIP
+        >>> preds = ["hello there", "general kenobi"]
+        >>> target = ["hello there", "master kenobi"]
+        >>> bert_score(preds=preds, target=target, lang="en")  # doctest: +SKIP
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:556]:
```suggestion
        >>> bert_score(preds, target, lang="en")  # doctest: +SKIP
```

File: torchmetrics/functional/text/bert.py (hunk 1 above)
Comment (stancld) [reply torchmetrics/functional/text/bert.py:457]: Ohh, that's true. I'm gonna fix that :]

**Comment by Borda** [comment]
@stancld is it ready? :]

**Comment by stancld** [comment]
> @stancld is it ready? :]

@Borda I need to find out the way of how to comply with `mypy` when `preds` and `predictions` are both `None` by default.. 🤔

**Comment by Borda** [comment]
> > @stancld is it ready? :]
> 
> @Borda I need to find out the way of how to comply with `mypy` when `preds` and `predictions` are both `None` by default.. thinking

I would set the line as ignore and add a todo note to remove the ignore after we remove this deprecation
there is no need to waste too much tie on finding a solution for something which is not critical and also just for one release :ice_cream:

**Comment by stancld** [comment]
> > > @stancld is it ready? :]
> > 
> > 
> > @Borda I need to find out the way of how to comply with `mypy` when `preds` and `predictions` are both `None` by default.. thinking
> 
> I would set the line as ignore and add a todo note to remove the ignore after we remove this deprecation there is no need to waste too much tie on finding a solution for something which is not critical and also just for one release 🍨

I was afraid of suggesting that :D But I agree it's not worthy to overcomplicate it for now. I'll add ignore statements and open an issue so that we will remove these statements afterwards.

File: torchmetrics/functional/text/bert.py [hunk 4]
```diff
@@ -530,14 +533,22 @@ def bert_score(
...
         baseline_url:
             A url path to the user's own  csv/tsv file with the baseline scale.
+        predictions:
+            Either an iterable of predicted sentences or a `Dict[str, torch.Tensor]` containing `input_ids` and
+            `attention_mask` `torch.Tensor`.
+            This argument is deprecated in v0.7 and will be removed in v0.8. Use `preds` instead.
+        references:
+            Either an iterable of target sentences or a `Dict[str, torch.Tensor]` containing `input_ids` and
+            `attention_mask` `torch.Tensor`.
+            This argument is deprecated in v0.7 and will be removed in v0.8. Use `target` instead.
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:543]:
```suggestion
        predictions:
            .. deprecated:: v0.7
                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
        references:
            .. deprecated:: v0.7
                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
```
just need to check therendering

File: torchmetrics/functional/text/bert.py [hunk 5]
```diff
@@ -530,14 +533,22 @@ def bert_score(
...
 
     Returns:
         Python dictionary containing the keys `precision`, `recall` and `f1` with corresponding values.
 
     Raises:
         ValueError:
-            If `len(predictions) != len(references)`.
-        ModuleNotFoundError:
+            If `len(preds) != len(target)`.
+        ValueError:
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:551]:
```suggestion
        ModuleNotFoundError:
```
or any reason for reverting it to `ValueError`?

File: torchmetrics/functional/text/bert.py [hunk 6]
```diff
@@ -548,14 +559,36 @@ def bert_score(
...
+        >>> bert_score(preds, target, lang="en")  # doctest: +SKIP
         {'precision': [0.99..., 0.99...],
          'recall': [0.99..., 0.99...],
          'f1': [0.99..., 0.99...]}
     """
-    if len(predictions) != len(references):
+    if preds is None and predictions is None:
+        raise ValueError("Either `preds` or `predictions` must be provided.")
+    if target is None and references is None:
+        raise ValueError("Either `target` or `references` must be provided.")
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:572]: :+1:

File: torchmetrics/functional/text/bleu.py [hunk 7]
```diff
@@ -139,38 +137,46 @@ def _bleu_score_compute(
...
         n_gram:
             Gram value ranged from 1 to 4 (Default 4)
         smooth:
             Whether or not to apply smoothing – see [2]
+        translate_corpus:
+            An iterable of machine translated corpus
+            This argument is deprecated in v0.7 and will be removed in v0.8. Use `preds` instead.
+        reference_corpus:
+            An iterable of iterables of reference corpus
+            This argument is deprecated in v0.7 and will be removed in v0.8. Use `target` instead.
```
Comment (Borda) [inline torchmetrics/functional/text/bleu.py:170]: the same with `.. deprecated::` probably in all other cases too

File: torchmetrics/functional/text/sacre_bleu.py [hunk 8]
```diff
@@ -278,20 +278,22 @@ def _lower(line: str, lowercase: bool) -> str:
...
-    translate_corpus: Sequence[str],
-    reference_corpus: Sequence[Sequence[str]],
+    preds: Union[None, Sequence[str]] = None,
+    target: Union[None, Sequence[Sequence[str]]] = None,
     n_gram: int = 4,
     smooth: bool = False,
     tokenize: Literal["none", "13a", "zh", "intl", "char"] = "13a",
     lowercase: bool = False,
+    translate_corpus: Union[None, Sequence[str]] = None,
+    reference_corpus: Union[None, Sequence[Sequence[str]]] = None,
```
Comment (Borda) [inline torchmetrics/functional/text/sacre_bleu.py:288]:
```suggestion
    translate_corpus: Union[None, Sequence[str]] = None,  # ToDo: remove in v0.8
    reference_corpus: Union[None, Sequence[Sequence[str]]] = None,  # ToDo: remove in v0.8
```
could be useful to not forget...

**Review by Borda — COMMENTED** [review]
just update the deprecated args in docs, otherwise looks good to me :tada:

File: torchmetrics/functional/text/bert.py (hunk 5 above)
Comment (stancld) [reply torchmetrics/functional/text/bert.py:555]: That was changed by mistake. I'll revert that

File: torchmetrics/functional/text/bert.py [hunk 9]
```diff
@@ -449,9 +451,16 @@ def _rescale_metrics_with_baseline(
     return all_metrics[..., 0], all_metrics[..., 1], all_metrics[..., 2]
 
 
+@deprecated(
+    args_mapping={"predictions": "preds", "references": "target"},
+    target=True,
+    stream=logging.warning,
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:457]: I think we want to keep the default `DeprecationWarning` or #744

File: torchmetrics/functional/text/bert.py [hunk 10]
```diff
@@ -530,14 +539,20 @@ def bert_score(
             A path to the user's own local csv/tsv file with the baseline scale.
         baseline_url:
             A url path to the user's own  csv/tsv file with the baseline scale.
+        predictions:
+            .. deprecated:: v0.7
+                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
+        references:
+            .. deprecated:: v0.7
+                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:547]:
```suggestion
        predictions:
        
            .. deprecated:: v0.7
                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
                
        references:
        
            .. deprecated:: v0.7
                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
                
```
sorry for the misleading suggestion before...

File: torchmetrics/functional/text/bleu.py [hunk 11]
```diff
@@ -139,38 +139,51 @@ def _bleu_score_compute(
...
         n_gram:
             Gram value ranged from 1 to 4 (Default 4)
         smooth:
             Whether or not to apply smoothing – see [2]
+        translate_corpus:
+            .. deprecated:: v0.7
+                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
+        reference_corpus:
+            .. deprecated:: v0.7
+                This argument is deprecated in favor of  `preds` and will be removed in v0.8.
```
Comment (Borda) [inline torchmetrics/functional/text/bleu.py:177]: drop these if they are not in the func signature...

Comment (stancld) [reply torchmetrics/functional/text/bleu.py:177]: @Borda Ok, I'll remove these. And do you think we will go with `pyDeprecate` then? :]

cc: @SkafteNicki

File: torchmetrics/functional/text/bert.py [hunk 12]
```diff
@@ -449,9 +451,16 @@ def _rescale_metrics_with_baseline(
...
+    target=True,
+    stream=logging.warning,
+    deprecated_in="0.7",
+    remove_in="0.8",
+)
 def bert_score(
-    predictions: Union[List[str], Dict[str, Tensor]],
-    references: Union[List[str], Dict[str, Tensor]],
+    preds: Union[None, List[str], Dict[str, Tensor]] = None,
+    target: Union[None, List[str], Dict[str, Tensor]] = None,
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:463]: in such case None is not allowed any more :]

File: torchmetrics/functional/text/bert.py [hunk 13]
```diff
@@ -548,14 +563,14 @@ def bert_score(
...
-        >>> bert_score(predictions=predictions, references=references, lang="en")  # doctest: +SKIP
+        >>> preds = ["hello there", "general kenobi"]
+        >>> target = ["hello there", "master kenobi"]
+        >>> bert_score(preds, target, lang="en")  # doctest: +SKIP
         {'precision': [0.99..., 0.99...],
          'recall': [0.99..., 0.99...],
          'f1': [0.99..., 0.99...]}
     """
-    if len(predictions) != len(references):
+    if len(preds) != len(target):  # type: ignore
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:573]: also if none is allowed this ignore is also not needed

File: torchmetrics/functional/text/bleu.py (hunk 11 above)
Comment (SkafteNicki) [reply torchmetrics/functional/text/bleu.py:177]: Agree that we can use `pyDeprecate` temporary for this

File: torchmetrics/functional/text/bert.py (hunk 9 above)
Comment (Borda) [reply torchmetrics/functional/text/bert.py:457]: @stancld ^^

Comment (stancld) [reply torchmetrics/functional/text/bert.py:457]: Will have a look :]

**Review by Borda — COMMENTED** [review]
just the warning stream, otherwise it looks good

**Comment by stancld** [comment]
> just the warning stream, otherwise it looks good

@Borda Actually, I don't know why, but I don't get any warnings with a default stream value 🤔

File: torchmetrics/functional/text/bert.py [hunk 14]
```diff
@@ -449,9 +451,16 @@ def _rescale_metrics_with_baseline(
     return all_metrics[..., 0], all_metrics[..., 1], all_metrics[..., 2]
 
 
+@deprecated(
+    args_mapping={"predictions": "preds", "references": "target"},
+    target=True,
+    stream=partial(warn, category=FutureWarning),
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:457]: lets remove this argument for now as it shall be done eventually consistently for all, so we can make it as separate PR

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]
